### DISCUSSION TITLE
Trouble with update template on server-side after receiving data from API

### DISCUSSION THREAD

**Discussion by Drider** [nuxt/nuxt#16991] [Questions] [answered]
Hi! I ran into a problem with update template on server-side after change shared state data. 
I have layout with side menu in app.vue and some pages. Also I create shared state with useState composable. When I make request on the page through useAsyncData and change data in shared state this changes does not trigger computed property which depend on shared state and changes in app.vue template. 

I debug the code and saw that first the code in the app.vue (or in component that place here) is processed, and then only on a page components. When the shared state changes after receiving data on some page, that not affect app.vue on server-side and on the client I get an warn  "[Vue warn]: Hydration text content mismatch.".

How is correct way to apply changes after make api request on server-side ?

Here is code example on [StackBlitz ](https://stackblitz.com/edit/github-r6khd4-an4xuf?file=app.vue)

**Comment by danielroe** [comment, marked as answer]
On the server, a Vue app is not reactive. So a change made to data within a page cannot affect the layout. You should ensure if you are fetching data that it happens 'above' or 'before' the component is rendered that depends on the data. A good place to do this is within `app.vue` or a plugin or a route middleware.

**Reply by Drider** [reply]
@danielroe thanks for the answer. Yes, I saw the description that the view works without reactivity on the server side. It would be convenient to encapsulate logic within pages. I'll try to find a different approach to requesting the necessary data that may affect the layout following your advice.
